state/util.go

```diff
@@ -15,7 +15,11 @@ func (m OidToIdxMap) Put(dbOid, objOid Oid, idx int32) {
 
 func (m OidToIdxMap) Get(dbOid, objOid Oid) int32 {
 	if _, ok := m[dbOid]; !ok {
-		return 0
+		return -1
```

Comment:
Other than finding it hard to believe I wrote something this bogus in the first place, the patch looks good.

Reply:
I think I wrote the original problem code here in https://github.com/pganalyze/collector/pull/61, you just expanded on it :-)